scripts/lf_build.py

```diff
@@ -91,4 +94,195 @@ def do_run(self, args, unknown_args):
+            print("Error: Use `--conf-overlays` option to pass config overlays to west")
+
+        federateNames = [name for name in os.listdir("./"+srcGenPath)]
+        
+        # Extract board name from west command
+        westCmds = args.west_commands.split(" ")
+        if "-b" in westCmds:
+            boardName = westCmds[westCmds.index("-b") + 1]
+        elif "--board" in westCmds:
+            boardName = westCmds[westCmds.index("--board") + 1]
+        else:
+            print("Error: board must be specified for lf-fed-build command.")
+
+        # Copy project configuration files into correct folders
+        for fedName in federateNames:
+            # Skip federate if federates to compile have been specified, and this federate is not included
+            if args.federate and fedName not in args.federate:
+                continue
+
+            userConfigPaths=f"prj_{fedName}.conf"
+            res = subprocess.Popen(f"cp {appPath}/prj_{fedName}.conf {srcGenPath}/{fedName}/", shell=True)
+            ret = res.wait()
+            if ret != 0:
+                print(f"Warning: Could not find federate config file {userConfigPaths}, using default config file only. \
+                      If there are multiple board federates, then the IP addresses might not be configured correctly.")
+                userConfigPaths=""
+
+            res = subprocess.Popen(f"cp {appPath}/{boardName}_{fedName}.overlay {srcGenPath}/{fedName}/boards/{boardName}.overlay", shell=True)
+            ret = res.wait()
+            if ret != 0:
+                print(f"Warning: Could not find federate dts overlay file. If there are multiple board federates, \
+                      then the board MAC addresses might become equal.")
+                
+            if args.conf_overlays:
+                res = subprocess.Popen(f"cp {appPath}/{args.conf_overlays} {srcGenPath}/{fedName}/", shell=True)
+                userConfigPaths += f" {args.conf_overlays}"
+
+            res = subprocess.Popen(f"cp {appPath}/Kconfig {srcGenPath}/{fedName}/", shell=True)
+            ret = res.wait()
+            if ret != 0:
+                exit(1)
+
+            compileDefs = ""
+            with open(f"{srcGenPath}/{fedName}/CompileDefinitions.txt") as f:
+                for line in f:
+                    line = line.replace("\n", "")
+                    compileDefs += f"-D{line} "
+                        
+            print(compileDefs)
+
+            # Invoke west in the `src-gen` directory. Pass in 
+            westCmd = f"west build {srcGenPath}/{fedName} --build-dir ./zephyr-{fedName}-build {args.west_commands} -- -DCMAKE_BUILD_TYPE=Test -DOVERLAY_CONFIG=\"{userConfigPaths}\" {compileDefs}"
+            print(f"Executing west command: `{westCmd}`")
+            res = subprocess.Popen(westCmd, shell=True)
+            ret = res.wait()
+            if ret != 0:
+                exit(1)
+
+class LfFedFlash(WestCommand):
```

Comment:
This command is very NXP EVK specific as it is using one particular flash/debug tool (pyOCD). As it is, I think it should be renamed to flash-evk or something. Alternatively we should look into how to make it more board independent.